sklearn/metrics/cluster/tests/test_unsupervised.py

```diff
@@ -15,29 +15,38 @@
 )
 from sklearn.metrics.cluster._unsupervised import _silhouette_reduce
 from sklearn.utils._testing import assert_array_equal
+from sklearn.utils.fixes import (
+    CSC_CONTAINERS,
+    CSR_CONTAINERS,
+    DOK_CONTAINERS,
+    LIL_CONTAINERS,
+)
 
 
 def test_silhouette():
     # Tests the Silhouette Coefficient.
     dataset = datasets.load_iris()
-    X_dense = dataset.data
-    X_csr = csr_matrix(X_dense)
-    X_csc = csc_matrix(X_dense)
-    X_dok = dok_matrix(X_dense)
-    X_lil = lil_matrix(X_dense)
-    y = dataset.target
+    X, y = dataset.data, dataset.target
+
+    for sparse_container in [
+        None,
+        *CSR_CONTAINERS,
+        *CSC_CONTAINERS,
+        *DOK_CONTAINERS,
+        *LIL_CONTAINERS,
+    ]:
+        if sparse_container is not None:
+            X = sparse_container(X)
```

Comment:
Since this is a loop, we can parametrize the function most probably. Could you use the concatenation instead of unpacking in this case.